robot/src/robot.py

```diff
@@ -18,9 +20,53 @@ def __init__(self):
 
         self.ultrasonic_sensor = UltrasonicSensor(Port.S4)
 
-    def obstacle_in_front(self):
+        self.routes = {1: [Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'),
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}',
+                           Event("MoveDistance", '{argument: 300}'), Event("Rotate", '{argument: 90}'))]}
+
+        self.event_source = event_source
+
+
+    async def set_movement_speed(self, event):
+        speed = event.data.get('argument')
+        self.drive_base.settings(speed=speed)
+
+    # Get the distance traveled by the robot in centimeters
+    async def distance_travelled(self):
+        return self.drive_base.distance() / 10
```

Comment:
Are we using centimetres for any particular reason? Do we not have the accuracy required for millimetres?